sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/EntraCommunicationTokenCredentialOptions.java

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.common;
+
+import com.azure.core.credential.TokenCredential;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static com.azure.communication.common.EntraCommunicationTokenUtils.allScopesStartWith;
+import static com.azure.communication.common.EntraCommunicationTokenUtils.TEAMS_EXTENSION_SCOPE_PREFIX;
+import static com.azure.communication.common.EntraCommunicationTokenUtils.COMMUNICATION_CLIENTS_SCOPE_PREFIX;
+import static com.azure.communication.common.EntraCommunicationTokenUtils.DEFAULT_SCOPE;
+import static com.azure.core.util.CoreUtils.isNullOrEmpty;
+
+/**
+ * The Entra Communication Token Options.
+ */
+public final class EntraCommunicationTokenCredentialOptions {
+
+    private final TokenCredential tokenCredential;
+    private final String resourceEndpoint;
+    private List<String> scopes;
+
+    /**
+     * Initializes a new instance of EntraCommunicationTokenCredentialOptions with default scopes.
+     *
+     * @param tokenCredential The credential capable of fetching an Entra user token.
+     * @param resourceEndpoint The URI of the Azure Communication Services resource. For example, https://myResource.communication.azure.com.
+     *
+     * @throws NullPointerException if tokenCredential is null.
+     * @throws IllegalArgumentException if resourceEndpoint is null or empty.
+     */
+    public EntraCommunicationTokenCredentialOptions(TokenCredential tokenCredential, String resourceEndpoint) {
+        Objects.requireNonNull(tokenCredential, "'tokenCredential' cannot be null.");
+        if (isNullOrEmpty(resourceEndpoint)) {
+            throw new IllegalArgumentException("'resourceEndpoint' cannot be null or empty.");
+        }
+        this.resourceEndpoint = resourceEndpoint;
+        this.tokenCredential = tokenCredential;
+        this.scopes = new ArrayList<String>() {
+            {
+                add(DEFAULT_SCOPE);
```

Comment:
maybe we can rename it CLIENTS_APP_DEFAULT_SCOPE or smth like this, so it would be readable which default scopes we are referring, wdyt?